src/Installer/finalizer/Program.cs

```diff
@@ -0,0 +1,262 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.Win32;
+using Microsoft.Win32.Msi;
+
+if (args.Length < 3)
+{
+    Console.WriteLine("Invalid arguments. Usage: finalizer <logPath> <sdkVersion> <platform>");
+    return;
+}
+
+string logPath = args[0];
+string sdkVersion = args[1];
+string platform = args[2];
+
+using StreamWriter logStream = new StreamWriter(logPath);
+
+Logger.Init(logStream);
+
+Logger.Log($"{nameof(logPath)}: {logPath}");
+Logger.Log($"{nameof(sdkVersion)}: {sdkVersion}");
+Logger.Log($"{nameof(platform)}: {platform}");
+
+try
+{
+    // Step 1: Parse and format SDK feature band version
+    string featureBandVersion = new SdkFeatureBand(sdkVersion).ToString();
+
+    // Step 2: Check if SDK feature band is installed
+    bool isInstalled = DetectSdk(featureBandVersion, platform);
+    if (isInstalled)
+    {
+        Logger.Log($"SDK with feature band {featureBandVersion} is already installed.");
+        return;
+    }
+
+    // Step 3: Remove dependent components if necessary
+    bool restartRequired = RemoveDependent(featureBandVersion);
```

Comment:
This should pass the dependent name, computed as `$"Microsoft.NET.Sdk,{featureBandVersion},{platform}` so you end up with something like `Microsoft.NET.Sdk,10.0.100,x64`

In the C code, it used to do this:

```
 // Create the dependent value, e.g., Microsoft.NET.Sdk,6.0.300,arm64
    hr = StrAllocFormatted(&sczDependent, L"Microsoft.NET.Sdk,%ls,%ls", sczFeatureBandVersion, sczPlatform);
```